activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -243,7 +243,10 @@ def owner_key_type
           end
 
           def reflection_scope
-            @reflection_scope ||= reflection.join_scopes(klass.arel_table, klass.predicate_builder, klass).inject(&:merge!) || klass.unscoped
+            @reflection_scope ||= [
+              klass.unscoped,
+              *reflection.join_scopes(klass.arel_table, klass.predicate_builder, klass)
+            ].inject(&:merge!)
```

Comment:
```suggestion
            @reflection_scope ||= begin
              reflection.join_scopes(klass.arel_table, klass.predicate_builder).inject(klass.unscoped, &:merge!)
            end
```

So I tested it on our app, and the regression we were seeing at the time don't come back. So I think this change wasn't a necessary part of the fix.

Please use the original version though, and I can merge.

Reply:
Thanks for having a look!
I actually didn't have a look at the previous version you mentioned. I just checked it and you're right - they basically behave the same.